server/src/test/java/keywhiz/service/daos/SecretControllerMockTest.java

```diff
@@ -0,0 +1,84 @@
+package keywhiz.service.daos;
+
+import java.util.Base64;
+import javax.swing.text.AbstractDocument;
+import keywhiz.KeywhizConfig;
+import keywhiz.service.crypto.ContentCryptographer;
+import keywhiz.service.crypto.SecretTransformer;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.ExpectedException;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+public class SecretControllerMockTest {
+  private static final Long MAXIMUM_SECRET_SIZE_IN_BYTES_INCLUSIVE = 100L;
+  private static final long NO_EXPIRY = 0;
+  private static final Long NO_MAXIMUM_SECRET_SIZE = null;
+
+  @Rule
+  public ExpectedException thrown = ExpectedException.none();
+
+  @Test
+  public void rejectsSecretWithSizeGreaterThanMaximumSecretSize() {
```

Comment:
A nice side effect of moving the validation logic from the DAO layer to the controller layer is that we don't have to deal with Jooq, making the test setup much simpler.